jdi-light-material-ui-tests/src/test/java/io/github/epam/material/tests/inputs/ButtonTests.java

```diff
@@ -28,137 +35,75 @@ public void before() {
-        iconLabelIcons.get(6).is().displayed();
+    public void enableTextButtonTest() {
+        defaultTextButton.label().is().text("DEFAULT");
+        defaultTextButton.click();
+        textButtonLastClick.is().text("Last click: Default");
     }
 
     @Test
-    public void iconButtonTest() {
-
-        iconButtons.get(1).click();
-        iconButtons.get(5).is().text("Last click: Delete enabled");
-        iconButtonIcons.get(1).is().displayed();
-
-        iconButtons.get(2).is().attr("disabled");
-        iconButtons.get(2).is().disabled();
-        iconButtonIcons.get(2).is().displayed();
-
-        iconButtons.get(3).click();
-        iconButtons.get(5).is().text("Last click: Alarm");
-        iconButtonIcons.get(3).is().displayed();
+    public void disableTextButtonTest() {
+        disabledTextButton.label().is().text("DISABLED");
+        disabledTextButton.is().disabled();
+        textButtonLastClick.is().text("Last click:");
+    }
 
-        iconButtons.get(4).click();
-        iconButtons.get(5).is().text("Last click: Shopping cart");
-        iconButtonIcons.get(4).is().displayed();
+    @Test
+    public void enableIconLabelButtonTest() {
+        deleteIconLabelButton.icon().is().displayed();
+        deleteIconLabelButton.label().is().text("DELETE");
+        deleteIconLabelButton.click();
+        iconLabelButtonLastClick.is().text("Last click: Delete");
     }
 
     @Test
-    public void customizedButtonTest() {
+    public void disableIconLabelButtonTest() {
+        talkIconLabelButton.icon().is().displayed();
+        talkIconLabelButton.label().is().text("TALK");
+        talkIconLabelButton.is().disabled();
+        iconLabelButtonLastClick.is().text("Last click:");
+    }
 
-        customizedButtons.get(1).click();
-        customizedButtons.get(1).is().text("CUSTOM CSS");
-        customizedButtons.get(4).is().text("Last click: Custom CSS");
+    @Test
+    public void enableIconButtonTest() {
+        deleteIconButton.icon().is().displayed();
+        deleteIconButton.click();
+        iconButtonLastClick.is().text("Last click: Delete enabled");
+    }
 
-        customizedButtons.get(2).click();
-        customizedButtons.get(2).is().text("THEME PROVIDER");
-        customizedButtons.get(4).is().text("Last click: Theme Provider");
+    @Test
+    public void disableIconButtonTest() {
+        disabledIconButton.icon().is().displayed();
+        disabledIconButton.is().disabled();
+        iconButtonLastClick.is().text("Last click:");
+    }
 
-        customizedButtons.get(3).click();
-        customizedButtons.get(3).is().text("Bootstrap");
-        customizedButtons.get(4).is().text("Last click: Bootstrap");
+    @Test
+    public void customizeButtonTest() {
+        customCssCustomizedButton.label().is().text("CUSTOM CSS");
+        customCssCustomizedButton.click();
+        customizedButtonLastClick.is().text("Last click: Custom CSS");
```

Comment:
I know that we talked about splitting large tests into individual ones. But now it seems that there are too many similar tests that can be combined into one using a data provider. For example, enableContainedButtonTest, enableTextButtonTest, customizeButtonTest. What do you think about this?

Reply:
Sure I would be more fancy to use DP, but
The goal of these tests not to test the page but to show the usage of elements.